rust/private/rustc.bzl

```diff
@@ -198,26 +198,35 @@ def collect_deps(label, deps, proc_macro_deps, aliases, toolchain):
         build_info,
     )
 
-def get_linker_and_args(ctx, rpaths):
-    if (len(BAZEL_VERSION) == 0 or
-        versions.is_at_least("0.18.0", BAZEL_VERSION)):
-        user_link_flags = ctx.fragments.cpp.linkopts
-    else:
-        user_link_flags = depset(ctx.fragments.cpp.linkopts)
-
+def get_cc_toolchain(ctx):
     cc_toolchain = find_cpp_toolchain(ctx)
+
     kwargs = {
         "ctx": ctx,
     } if len(BAZEL_VERSION) == 0 or versions.is_at_least(
         "0.25.0",
         BAZEL_VERSION,
     ) else {}
+
     feature_configuration = cc_common.configure_features(
         cc_toolchain = cc_toolchain,
         requested_features = ctx.features,
         unsupported_features = ctx.disabled_features,
         **kwargs
     )
+
+    return cc_toolchain, feature_configuration
+
+def get_cc_user_link_flags(ctx):
+    if (len(BAZEL_VERSION) == 0 or
+        versions.is_at_least("0.18.0", BAZEL_VERSION)):
+        return ctx.fragments.cpp.linkopts
```

Comment:
opt: Drop support for Bazel < 1?

Reply:
I think in a followup so its more of an atomic commit and we can get all bazel versioning in one swoop

Reply:
Make sense